includes/admin/settings/register-settings.php

```diff
@@ -1374,6 +1379,51 @@ function edd_settings_sanitize_misc_accounting( $input ) {
 
 add_filter( 'edd_settings_misc-accounting_sanitize', 'edd_settings_sanitize_misc_accounting' );
 
+/**
+ * Default Tax Rate Settins Sanitization
+ *
+ * Saves to adjustments database as special type
+ *
+ * @since 3.0
+ *
+ * @param array $input The value inputted in the field
+ *
+ * @return array $input Sanitized value.
+ */
+function edd_settings_sanitize_tax_rate( $input ) {
+
+	if ( ! current_user_can( 'manage_shop_settings' ) ) {
+		return $input;
+	}
+
+	if ( ! isset( $_POST['edd_settings']['tax_rate'] ) ) {
+		return $input;
+	}
+
+	$tax_rate = $_POST['edd_settings']['tax_rate'];
+
+	$adjustment_data = array(
+		'name'        => __( 'Global Rate', 'easy-digital-dowloads' ),
+		'type'        => 'tax_rate',
+		'scope'       => 'global',
+		'amount_type' => 'percent',
+		'amount'      => floatval( $tax_rate ),
+	);
+
+	$id = edd_get_option( 'edd_default_tax_rate_id' );
+
+	if ( ! empty( $id ) ) {
+		edd_update_adjustment( $id, $adjustment_data );
```

Comment:
@mintplugins

I can see why you would say that. Although, I'm confused because an an existing function "edd_settings_sanitize_taxes" updates through the same function I'm using here. Both edd_add_adjustment && edd_update_adjustment.

Maybe I'm missing something?

Thanks.

Reply:
While I agree with @mintplugins that we should avoid making DB changes in sanitization functions, we do have to sometimes make concessions. Due to the tax settings being injected into the super-non-flexible WordPress settings API, this is how we've always saved taxes. It'd be nice to make everything perfect in 3.0 but I don't see this as a big enough evil to necessitate re-engineering how we save tax rates in the settings screen.